net/golioth/system_client.c

```diff
@@ -171,6 +189,51 @@ static int init_tls(void)
 	return 0;
 }
 
+static int init_tls_auth_cert(void)
+{
+	int err;
+
+	err = tls_credential_add(CONFIG_GOLIOTH_SYSTEM_CLIENT_CREDENTIALS_TAG,
+				 TLS_CREDENTIAL_CA_CERTIFICATE,
+				 ca_certificate,
+				 ARRAY_SIZE(ca_certificate));
+	if (err < 0) {
+		LOG_ERR("Failed to register CA cert: %d", err);
+		return err;
+	}
+
+	err = tls_credential_add(CONFIG_GOLIOTH_SYSTEM_CLIENT_CREDENTIALS_TAG,
+				 TLS_CREDENTIAL_SERVER_CERTIFICATE,
+				 tls_server_certificate,
```

Comment:
I understand that Zephyr calls it `TLS_CREDENTIAL_SERVER_CERTIFICATE`, but isn't this actually the public client certificate that we send to the server? Just double-checking my understand.

Reply:
This name is confusing! Yeah, it is public client cert. I'll rename `tls_server_certificate` array to `tls_client_crt` to make it more consistent with Kconfig options, default file names, etc.